### PR TITLE
SRE-250 support nomad access to google artifact registry

### PR DESCRIPTION
**Pull request by tundeaoni**

### CODE REVIEW

File: tasks/assert.yml [hunk 1]
```diff
@@ -132,4 +132,11 @@
       - nomad_client_host_networks is iterable
     quiet: true
   when:
-    - nomad_client
+    - nomad_client | bool
```
Comment (hollow) [inline tasks/assert.yml]: `| bool` not needed

File: defaults/main.yml [hunk 2]
```diff
@@ -33,3 +33,5 @@ nomad_client: false
 nomad_client_node_class: worker
 nomad_client_host_volumes: []
 nomad_client_host_networks: []
+nomad_docker_auth_enabled: false
+nomad_docker_auth_file: /root/.docker/config.json
```
Comment (hollow) [inline defaults/main.yml:37]: This var is not asserted anymore

File: tasks/main.yml [hunk 3]
```diff
@@ -79,6 +79,13 @@
   notify:
     - Restart nomad
 
+- name: Store docker auth config
+  ansible.builtin.copy:
+    content: "{{ nomad_docker_auth_config }}"
+    dest: "{{ nomad_docker_auth_file }}"
```
Comment (hollow) [inline tasks/main.yml:85]: Always specify `owner`, `group`, and `mode` for all files

File: templates/client.hcl.j2 [hunk 4]
```diff
@@ -29,6 +29,11 @@ client {
 
 plugin "docker" {
   config {
+    {% if nomad_docker_auth_enabled is sameas true %}
```
Comment (hollow) [inline templates/client.hcl.j2]: Fix stray whitespace

**Review by hollow — CHANGES_REQUESTED** [review]

File: defaults/main.yml (hunk 2 above)
Comment (tundeaoni) [reply defaults/main.yml:37]: `- name: Test if nomad_docker_auth_config is set correctly
  ansible.builtin.assert:
    that:
      - nomad_docker_auth_config is defined
  when:
    - nomad_docker_auth_enabled | bool`
    
    and nomad_docker_auth_enabled defaults to `false`

Comment (hollow) [reply defaults/main.yml:37]: `nomad_docker_auth_file` != `nomad_docker_auth_config`

**Pull request merged**